cpp/include/raft/neighbors/ann_mg.cuh

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#ifndef RAFT_EXPLICIT_INSTANTIATE_ONLY
+#include "ann_mg-inl.cuh"
+#endif
+
+#ifdef RAFT_COMPILED
+#include "ann_mg-ext.cuh"
+#endif
```

Comment:
Thanks @viclafargue for working on this. Just a quick comment on header organization.

I am not convinced that we need to complicate the interface with separate `-inl` and `-ext` headers. According to our [developer guide](https://github.com/rapidsai/raft/blob/branch-23.12/docs/source/developer_guide.md#header-organization-of-expensive-function-templates):
>Any header file that defines an expensive function template (say expensive.cuh) should be split in three parts: expensive.cuh, expensive-inl.cuh, and expensive-ext.cuh

I do not think that we are defining expensive to compile functions here.  I believe we will simply wrap the existing ANN methons and dispatch them onto multiple devices. I expect the code here to be relatively quick to compile, and therfore we could stay with a simple header structure.